# ARTIFICIAL NEURAL NETWORK CLASSIFICATION MODEL


In [0]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [0]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [0]:
df=pd.read_csv('/content/drive/My Drive/Data Mining in Science/praktikum/3. Klasifikasi/iris_species.csv').dropna()
df.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

# Data Preprocessing

In [0]:
X = df.iloc[:, :-1].values
y = df.iloc[:, 5].values

In [0]:
#Creating numeric classes for the dependent variable, y (Species)

from sklearn.preprocessing import LabelEncoder
y = LabelEncoder().fit_transform(y)
y

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

In [0]:
# Splitting the dataset into the Training set and Test set; working on 75% training set

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.25, random_state = 0)

In [0]:
# Feature Scaling

from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

# Creating the ANN Model


In [0]:
import keras
from keras.models import Sequential
from keras.layers import Dense

# Initialising the ANN
classifier = Sequential()

# Adding the input layer and the first hidden layer
classifier.add(Dense(output_dim = 3, init = 'uniform', activation = 'relu', input_dim = 5))

# Adding the second hidden layer
classifier.add(Dense(output_dim = 3, init = 'uniform', activation = 'relu'))

# Adding the output layer
classifier.add(Dense(output_dim = 1, init = 'uniform', activation = 'sigmoid'))

# Compiling the ANN
classifier.compile(optimizer = 'adam', loss = 'binary_crossentropy', metrics = ['accuracy'])

# Fitting the ANN to the Training set
classifier.fit(X_train, y_train, batch_size = 10, nb_epoch = 50)


/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:9: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(activation="relu", input_dim=5, units=3, kernel_initializer="uniform")`
  if __name__ == '__main__':
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:12: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(activation="relu", units=3, kernel_initializer="uniform")`
  if sys.path[0] == '':
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:15: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(activation="sigmoid", units=1, kernel_initializer="uniform")`
  from ipykernel import kernelapp as app
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:21: UserWarning: The `nb_epoch` argument in `fit` has been renamed `epochs`.


Epoch 1/50
112/112 [==============================] - 0s 3ms/step - loss: 0.6901 - acc: 0.3036
Epoch 2/50
112/112 [==============================] - 0s 166us/step - loss: 0.6838 - acc: 0.3036
Epoch 3/50
112/112 [==============================] - 0s 134us/step - loss: 0.6771 - acc: 0.3036
Epoch 4/50
112/112 [==============================] - 0s 128us/step - loss: 0.6694 - acc: 0.3036
Epoch 5/50
112/112 [==============================] - 0s 143us/step - loss: 0.6610 - acc: 0.3036
Epoch 6/50
112/112 [==============================] - 0s 127us/step - loss: 0.6519 - acc: 0.3036
Epoch 7/50
112/112 [==============================] - 0s 142us/step - loss: 0.6409 - acc: 0.3036
Epoch 8/50
112/112 [==============================] - 0s 138us/step - loss: 0.6271 - acc: 0.3036
Epoch 9/50
112/112 [==============================] - 0s 127us/step - loss: 0.6108 - acc: 0.3036
Epoch 10/50
112/112 [==============================] - 0s 130us/step - loss: 0.5905 - acc: 0.3036
Epoch 11/50
112/112 [==========

The accuracy converged at 0.6339
Therefore, the ANN classification model is able to predict the out variable, y, with a 63.39% accuracy on the training set

# Making the predictions and evaluating the model - applying ANN model on the test set


In [0]:
# Predicting the Test set results
# Where the model is able to predict the output with a probability greater than 0.5, we make it true; otherwise, false

y_pred = classifier.predict(X_test)
y_pred = (y_pred > 0.5)
y_pred

array([[ True],
       [ True],
       [False],
       [ True],
       [False],
       [ True],
       [False],
       [ True],
       [ True],
       [ True],
       [ True],
       [ True],
       [ True],
       [ True],
       [ True],
       [False],
       [ True],
       [ True],
       [False],
       [False],
       [ True],
       [ True],
       [False],
       [False],
       [ True],
       [False],
       [False],
       [ True],
       [ True],
       [False],
       [ True],
       [ True],
       [False],
       [ True],
       [ True],
       [ True],
       [False],
       [ True]])

# Evaluating Model Prediction Accuracy

In [0]:
correct_prediction = 0

sample_space = y_pred.shape[0]

for result in y_pred:
  
    if result == True:
        correct_prediction += 1
        
accuracy = correct_prediction/sample_space*100

print("Total Sample: {}".format(sample_space))
print("\nCorrect Predictions: {}".format(correct_prediction))
print("\nAccuracy: {}".format(accuracy))

Total Sample: 38

Correct Predictions: 25

Accuracy: 65.78947368421053


The ANN model is able to obtain a prediction accuracy of 65.79% on the test set
